src/foam/nanos/dig/drivers/DigFormatDriver.js

```diff
@@ -178,10 +178,11 @@ foam.CLASS({
 
       long pageSize = DigFormatDriver.MAX_PAGE_SIZE;
       if ( ! SafetyUtil.isEmpty(limit) ) {
+        AuthService auth = (AuthService) x.get("auth");
         long l = Long.valueOf(limit);
-        if ( l == 0 ) {
+        
+        if ( l == 0 && auth.check(x, "service.dig.read-all-records") ) {
```

Comment:
I think the && part should be in a nested 'if' so you don't trigger the 'else if' below when the user doesn't have the permission and then set the limit to 0.